File: utils/units.go
// Created by DINKIssTyle on 2026.
// Copyright (C) 2026 DINKI'ssTyle. All rights reserved.

package utils

const (
	// ScreenDPI 화면 표시용 기본 DPI
	ScreenDPI = 96.0
	// PrintDPI 인쇄용 DPI
	PrintDPI = 300.0
	// MMPerInch 1인치당 밀리미터 비율
	MMPerInch = 25.4
)

// MMToInch converts mm to inches
func MMToInch(mm float64) float64 {
	return mm / MMPerInch
}

// InchToMM converts inches to mm
func InchToMM(inch float64) float64 {
	return inch * MMPerInch
}

// MMToPx converts mm to pixels
func MMToPx(mm float64, dpi float64) float64 {
	if dpi == 0 {
		dpi = ScreenDPI
	}
	return mm / MMPerInch * dpi
}

// InchToPx converts inches to pixels
func InchToPx(inch float64, dpi float64) float64 {
	if dpi == 0 {
		dpi = ScreenDPI
	}
	return inch * dpi
}

// PxToMM converts pixels to mm
func PxToMM(px float64, dpi float64) float64 {
	if dpi == 0 {
		dpi = ScreenDPI
	}
	return px / dpi * MMPerInch
}

// PxToInch converts pixels to inches
func PxToInch(px float64, dpi float64) float64 {
	if dpi == 0 {
		dpi = ScreenDPI
	}
	return px / dpi
}

// ToPx converts a value of given unit to pixels
func ToPx(value float64, unit string, dpi float64) float64 {
	if dpi == 0 {
		dpi = ScreenDPI
	}
	switch unit {
	case "mm":
		return float64(MMToPx(value, dpi))
	case "inch":
		return float64(InchToPx(value, dpi))
	default:
		return value
	}
}

// FromPx converts a pixel value to the given unit
func FromPx(px float64, unit string, dpi float64) float64 {
	if dpi == 0 {
		dpi = ScreenDPI
	}
	switch unit {
	case "mm":
		return PxToMM(px, dpi)
	case "inch":
		return PxToInch(px, dpi)
	default:
		return px
	}
}

// MMToPt converts mm to points (1pt = 1/72 inch)
func MMToPt(mm float64) float64 {
	return mm / MMPerInch * 72
}

// PtToMM converts points to mm
func PtToMM(pt float64) float64 {
	return pt / 72 * MMPerInch
}

File: _archive/utils/units.py
# Created by DINKIssTyle on 2026.
# Copyright (C) 2026 DINKI'ssTyle. All rights reserved.

"""단위 변환 유틸리티 (mm, inch, px)"""


# 화면 표시용 기본 DPI
SCREEN_DPI = 96
# 인쇄용 DPI
PRINT_DPI = 300

MM_PER_INCH = 25.4


def mm_to_inch(mm: float) -> float:
    return mm / MM_PER_INCH


def inch_to_mm(inch: float) -> float:
    return inch * MM_PER_INCH


def mm_to_px(mm: float, dpi: float = SCREEN_DPI) -> float:
    return mm / MM_PER_INCH * dpi


def inch_to_px(inch: float, dpi: float = SCREEN_DPI) -> float:
    return inch * dpi


def px_to_mm(px: float, dpi: float = SCREEN_DPI) -> float:
    return px / dpi * MM_PER_INCH


def px_to_inch(px: float, dpi: float = SCREEN_DPI) -> float:
    return px / dpi


def to_px(value: float, unit: str, dpi: float = SCREEN_DPI) -> float:
    """주어진 단위의 값을 픽셀로 변환"""
    if unit == "mm":
        return mm_to_px(value, dpi)
    elif unit == "inch":
        return inch_to_px(value, dpi)
    return value


def from_px(px: float, unit: str, dpi: float = SCREEN_DPI) -> float:
    """픽셀 값을 주어진 단위로 변환"""
    if unit == "mm":
        return px_to_mm(px, dpi)
    elif unit == "inch":
        return px_to_inch(px, dpi)
    return px


def mm_to_pt(mm: float) -> float:
    """mm를 포인트(pt)로 변환 (1pt = 1/72 inch)"""
    return mm / MM_PER_INCH * 72


def pt_to_mm(pt: float) -> float:
    """포인트(pt)를 mm로 변환"""
    return pt / 72 * MM_PER_INCH
